core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -416,6 +417,12 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     }
   }
 
+  protected def removeExecutorAsync(
```

Comment:
Any reason why we can't just make `removeExecutor` always use the async API?

It doesn't look like any of the call sites actually relies on it being synchronous.